python/tvm/relax/vm.py

```diff
@@ -585,6 +585,9 @@ def foo(x: Tensor((3, 4), "float32"), y: Tensor((3, 4), "float32")):
     passes.append(relax.transform.RewriteDataflowReshape())
     passes.append(relax.transform.ToNonDataflow())
     passes.append(relax.transform.CallTIRRewrite())
+    if target.kind.name == "cuda":
+        passes.append(relax.transform.CanonicalizeBindings())
```

Comment:
One comment: @vinx13 would be good to explain why CanonicalizeBindings is needed here

Reply:
CallTirRewrite will add redundant bindings. 
```
%1 = alloc_tensor
call_tir(%1, ..)
%2 = %1
```
That makes dep analysis harder. We can improve this